homeassistant/components/yolink/__init__.py

```diff
@@ -62,11 +62,16 @@ def on_message_callback(message: tuple[str, BRDP]):
         resolved_state = data.data
         if resolved_state is None:
             return
-        device_coordinators = hass.data[DOMAIN][ATTR_COORDINATORS]
-        if device_coordinators is not None:
-            device_coordinator = device_coordinators.get(device_id)
-            if device_coordinator is not None:
-                device_coordinator.async_set_updated_data(resolved_state)
+        entry_data = hass.data[DOMAIN].get(entry.entry_id)
+        if entry_data is None:
+            return None
+        device_coordinators = entry_data.get(ATTR_COORDINATORS)
+        if device_coordinators is None:
+            return None
+        device_coordinator = device_coordinators.get(device_id)
+        if device_coordinator is None:
+            return None
+        return device_coordinator.async_set_updated_data(resolved_state)
```

Comment:
If this doesn't return anything, we should not return the return value of this function.
```suggestion
        device_coordinator.async_set_updated_data(resolved_state)
```